homeassistant/components/wink.py

```diff
@@ -41,13 +56,39 @@ def setup(hass, config):
     return True
 
 
-class WinkToggleDevice(ToggleEntity):
-    """Represents a Wink toggle (switch) device."""
+class WinkDevice(Entity):
+    """Represents a base Wink device."""
 
     def __init__(self, wink):
         """Initialize the Wink device."""
+        from pubnub import Pubnub
         self.wink = wink
         self._battery = self.wink.battery_level
+        if (SUBSCRIPTION_HANDLER is not None and
+                self.wink.pubnub_channel is not None):
+            self.sub_key = self.wink.pubnub_key
+            self.sub_channel = self.wink.pubnub_channel
+            if self.sub_channel in CHANNELS:
+                pubnub = Pubnub(subscribe_key=self.sub_key, publish_key="N/A")
```

Comment:
This object is for powerstips and the multi sensor spotter. These devices have two or more entities in home-assistant but all of those entities that are part of the same physical device share the same channel. Without this check whenever a new entity was added with the same channel it would replace the previous entry. This check creates a new subscription handler so each entity has its own connection to pubnub.